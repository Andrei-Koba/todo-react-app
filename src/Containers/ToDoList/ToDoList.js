import React from 'react';
import { useSelector } from 'react-redux';
import styles from './ToDoList.module.scss';
import ControlPannel from '../ControlPannel/ControlPannel';
import ToDoItem from '../ToDoItem/ToDoItem';
import filterFunction from '../../Redux/filterFunction';

export default function ToDoList() {
  const allToDoItems = useSelector((state) => state.todo);
  const todoItems = useSelector((state) => filterFunction(state));

  const itemsContainers = todoItems.map((item) => <ToDoItem item={item} key={item.id} />);

  return (
    <div className={styles.todoList}>
      <ul>{itemsContainers}</ul>
      {allToDoItems.length > 0 && <ControlPannel />}
    </div>
  );
}
