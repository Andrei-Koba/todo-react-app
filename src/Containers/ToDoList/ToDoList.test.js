import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render, fireEvent } from '@testing-library/react';
import ToDoList from './ToDoList';

const state = {
  todo: [
    {
      id: 'testId1',
      text: 'testText1',
      isCompleted: true,
    },
    {
      id: 'testId2',
      text: 'testText2',
      isCompleted: false,
    },
    {
      id: 'testId3',
      text: 'testText3',
      isCompleted: true,
    },
    {
      id: 'testId4',
      text: 'testText4',
      isCompleted: false,
    },
  ],
};
const emptyState = {
  todo: [
    {
      id: 'testId1',
      text: 'testText1',
      isCompleted: true,
    },
    {
      id: 'testId2',
      text: 'testText2',
      isCompleted: false,
    },
    {
      id: 'testId3',
      text: 'testText3',
      isCompleted: true,
    },
    {
      id: 'testId4',
      text: 'testText4',
      isCompleted: false,
    },
  ],
};
const mockStore = configureStore([]);

describe('ToDoList', () => {
  it('renders correctly with items', () => {
    const store = mockStore(state);
    const component = render(
      <Provider store={store}>
        <ToDoList />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly without items', () => {
    const store = mockStore(emptyState);
    const component = render(
      <Provider store={store}>
        <ToDoList />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
