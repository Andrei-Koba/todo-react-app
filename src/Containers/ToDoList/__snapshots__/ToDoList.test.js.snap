// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList renders correctly with items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todoList"
      >
        <ul>
          <li
            class="todoItem "
          >
            <label
              class="checkbox"
              for="testId1"
            >
              <div>
                <div
                  class="styledCheckbox checked"
                />
                <input
                  checked=""
                  id="testId1"
                  type="checkbox"
                />
              </div>
              <span
                class="itemText crossed"
              >
                testText1
              </span>
            </label>
          </li>
          <li
            class="todoItem "
          >
            <label
              class="checkbox"
              for="testId2"
            >
              <div>
                <div
                  class="styledCheckbox unchecked"
                />
                <input
                  id="testId2"
                  type="checkbox"
                />
              </div>
              <span
                class="itemText undefined"
              >
                testText2
              </span>
            </label>
          </li>
          <li
            class="todoItem "
          >
            <label
              class="checkbox"
              for="testId3"
            >
              <div>
                <div
                  class="styledCheckbox checked"
                />
                <input
                  checked=""
                  id="testId3"
                  type="checkbox"
                />
              </div>
              <span
                class="itemText crossed"
              >
                testText3
              </span>
            </label>
          </li>
          <li
            class="todoItem "
          >
            <label
              class="checkbox"
              for="testId4"
            >
              <div>
                <div
                  class="styledCheckbox unchecked"
                />
                <input
                  id="testId4"
                  type="checkbox"
                />
              </div>
              <span
                class="itemText undefined"
              >
                testText4
              </span>
            </label>
          </li>
        </ul>
        <div
          class="controlPannel"
        >
          <span
            class="counter"
          >
            4 items left
          </span>
          <div>
            <button
              class="filterButton  "
              name="All"
              type="button"
            >
              All
            </button>
            <button
              class="filterButton  "
              name="Active"
              type="button"
            >
              Active
            </button>
            <button
              class="filterButton  "
              name="Completed"
              type="button"
            >
              Completed
            </button>
          </div>
          <button
            class="clearButton "
            type="button"
          >
            Clear completed
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todoList"
    >
      <ul>
        <li
          class="todoItem "
        >
          <label
            class="checkbox"
            for="testId1"
          >
            <div>
              <div
                class="styledCheckbox checked"
              />
              <input
                checked=""
                id="testId1"
                type="checkbox"
              />
            </div>
            <span
              class="itemText crossed"
            >
              testText1
            </span>
          </label>
        </li>
        <li
          class="todoItem "
        >
          <label
            class="checkbox"
            for="testId2"
          >
            <div>
              <div
                class="styledCheckbox unchecked"
              />
              <input
                id="testId2"
                type="checkbox"
              />
            </div>
            <span
              class="itemText undefined"
            >
              testText2
            </span>
          </label>
        </li>
        <li
          class="todoItem "
        >
          <label
            class="checkbox"
            for="testId3"
          >
            <div>
              <div
                class="styledCheckbox checked"
              />
              <input
                checked=""
                id="testId3"
                type="checkbox"
              />
            </div>
            <span
              class="itemText crossed"
            >
              testText3
            </span>
          </label>
        </li>
        <li
          class="todoItem "
        >
          <label
            class="checkbox"
            for="testId4"
          >
            <div>
              <div
                class="styledCheckbox unchecked"
              />
              <input
                id="testId4"
                type="checkbox"
              />
            </div>
            <span
              class="itemText undefined"
            >
              testText4
            </span>
          </label>
        </li>
      </ul>
      <div
        class="controlPannel"
      >
        <span
          class="counter"
        >
          4 items left
        </span>
        <div>
          <button
            class="filterButton  "
            name="All"
            type="button"
          >
            All
          </button>
          <button
            class="filterButton  "
            name="Active"
            type="button"
          >
            Active
          </button>
          <button
            class="filterButton  "
            name="Completed"
            type="button"
          >
            Completed
          </button>
        </div>
        <button
          class="clearButton "
          type="button"
        >
          Clear completed
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ToDoList renders correctly without items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todoList"
      >
        <ul>
          <li
            class="todoItem "
          >
            <label
              class="checkbox"
              for="testId1"
            >
              <div>
                <div
                  class="styledCheckbox checked"
                />
                <input
                  checked=""
                  id="testId1"
                  type="checkbox"
                />
              </div>
              <span
                class="itemText crossed"
              >
                testText1
              </span>
            </label>
          </li>
          <li
            class="todoItem "
          >
            <label
              class="checkbox"
              for="testId2"
            >
              <div>
                <div
                  class="styledCheckbox unchecked"
                />
                <input
                  id="testId2"
                  type="checkbox"
                />
              </div>
              <span
                class="itemText undefined"
              >
                testText2
              </span>
            </label>
          </li>
          <li
            class="todoItem "
          >
            <label
              class="checkbox"
              for="testId3"
            >
              <div>
                <div
                  class="styledCheckbox checked"
                />
                <input
                  checked=""
                  id="testId3"
                  type="checkbox"
                />
              </div>
              <span
                class="itemText crossed"
              >
                testText3
              </span>
            </label>
          </li>
          <li
            class="todoItem "
          >
            <label
              class="checkbox"
              for="testId4"
            >
              <div>
                <div
                  class="styledCheckbox unchecked"
                />
                <input
                  id="testId4"
                  type="checkbox"
                />
              </div>
              <span
                class="itemText undefined"
              >
                testText4
              </span>
            </label>
          </li>
        </ul>
        <div
          class="controlPannel"
        >
          <span
            class="counter"
          >
            4 items left
          </span>
          <div>
            <button
              class="filterButton  "
              name="All"
              type="button"
            >
              All
            </button>
            <button
              class="filterButton  "
              name="Active"
              type="button"
            >
              Active
            </button>
            <button
              class="filterButton  "
              name="Completed"
              type="button"
            >
              Completed
            </button>
          </div>
          <button
            class="clearButton "
            type="button"
          >
            Clear completed
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todoList"
    >
      <ul>
        <li
          class="todoItem "
        >
          <label
            class="checkbox"
            for="testId1"
          >
            <div>
              <div
                class="styledCheckbox checked"
              />
              <input
                checked=""
                id="testId1"
                type="checkbox"
              />
            </div>
            <span
              class="itemText crossed"
            >
              testText1
            </span>
          </label>
        </li>
        <li
          class="todoItem "
        >
          <label
            class="checkbox"
            for="testId2"
          >
            <div>
              <div
                class="styledCheckbox unchecked"
              />
              <input
                id="testId2"
                type="checkbox"
              />
            </div>
            <span
              class="itemText undefined"
            >
              testText2
            </span>
          </label>
        </li>
        <li
          class="todoItem "
        >
          <label
            class="checkbox"
            for="testId3"
          >
            <div>
              <div
                class="styledCheckbox checked"
              />
              <input
                checked=""
                id="testId3"
                type="checkbox"
              />
            </div>
            <span
              class="itemText crossed"
            >
              testText3
            </span>
          </label>
        </li>
        <li
          class="todoItem "
        >
          <label
            class="checkbox"
            for="testId4"
          >
            <div>
              <div
                class="styledCheckbox unchecked"
              />
              <input
                id="testId4"
                type="checkbox"
              />
            </div>
            <span
              class="itemText undefined"
            >
              testText4
            </span>
          </label>
        </li>
      </ul>
      <div
        class="controlPannel"
      >
        <span
          class="counter"
        >
          4 items left
        </span>
        <div>
          <button
            class="filterButton  "
            name="All"
            type="button"
          >
            All
          </button>
          <button
            class="filterButton  "
            name="Active"
            type="button"
          >
            Active
          </button>
          <button
            class="filterButton  "
            name="Completed"
            type="button"
          >
            Completed
          </button>
        </div>
        <button
          class="clearButton "
          type="button"
        >
          Clear completed
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
